internal/llrp/device_service.go

```diff
@@ -295,3 +283,39 @@ func (ds DSClient) put(path string, data []byte) error {
 
 	return nil
 }
+
+// tryGet attempts to make an HTTP GET call to the device service at the specified path. It will
+// try up to maxTries times and sleeps sleepInterval*tryCount in-between tries. It will return the
+// raw response and error objects of the final HTTP call that was completed.
+// It determines a successful try as anything returning 2xx http code.
+// Note: if err is nil, caller is expected to close response body
+func (ds DSClient) tryGet(path string) (resp *http.Response, err error) {
```

Comment:
Like I mentioned in the other comment, this code and some others in this file are from PR #47 , so it is confusing to me why they are showing up as "new".

Reply:
> There is something strange going on with your branch. It shows as 20 commits ahead, 10 commits behind `main`. It also shows code that I added in PR #47 as new code from this PR for some reason. It also shows there are conflicts on this branch, which is strange, as it did not before.
> 
> I think somewhere along the road something got squashed in your branch and lost its history against the main repo. I even observed this effect when I tried to create my PR against your branch.

Recreated the PR by squashing it to just my changes